languages-and-frameworks/axum.html.markerb

```diff
@@ -0,0 +1,115 @@
+---
+title: "Run a Rust Axum App"
+layout: framework_docs
+sitemap: false
+order: 1
+---
+
+<% app_name = "hello-next" %>
+<%= partial "/docs/js/frameworks/partials/intro", locals: { runtime: "Rust Axum", link: "https://docs.rs/axum/latest/axum/" } %>
+
+Deploying an [Axum](https://docs.rs/axum/latest/axum/) app on fly.io is really straightforward! With the help of the wonderful [cargo chef](https://github.com/LukeMathWalker/cargo-chef) we get great build times and small images.
+
+## _Deploy an Axum App_
+
+<%= partial "/docs/js/frameworks/partials/flyctl" %>
+
+Next, if you don't already have an existing Axum application, you can create one quickly with the following command:
+
+```cmd
+cargo new axum-app
+cd axum-app
+```
+
+Then we have to add some dependencies to the project:
+
+```cmd
+cargo add axum
+cargo add tokio -F macros -F rt-multi-thread
+```
+
+Now, let's create a simple Axum app in `src/main.rs`:
+
+```rust
+use axum::{routing::get, Router};
+
+#[tokio::main]
+async fn main() {
+    let app = Router::new().route("/", get(|| async { "Hello, World!" }));
+    let listener = tokio::net::TcpListener::bind("0.0.0.0:8080").await.unwrap();
+    axum::serve(listener, app).await.unwrap();
+}
+```
+
+This will display a simple "Hello, World!" message when you visit the root URL.
+Take note that we serve the app on port `8080`.
+
+We can confirm everything works fine by running `cargo run` and checking out `http://localhost:8080`.
+
+Now we add a `Dockerfile` to the project:
+
+```dockerfile
+FROM lukemathwalker/cargo-chef:latest-rust-1 AS chef
+WORKDIR /app
+
+FROM chef AS planner
+COPY . .
+RUN cargo chef prepare --recipe-path recipe.json
+
+FROM chef AS builder 
+COPY --from=planner /app/recipe.json recipe.json
+# Build dependencies - this is the caching Docker layer!
+RUN cargo chef cook --release --recipe-path recipe.json
+# Build application
+COPY . .
+RUN cargo build --release --bin axum-app
+
+# We do not need the Rust toolchain to run the binary!
+FROM debian:bookworm-slim AS runtime
+WORKDIR /app
+COPY --from=builder /app/target/release/axum-app /usr/local/bin
+ENTRYPOINT ["/usr/local/bin/axum-app"] 
+```
+
+Modify the dockerfile to match your project name (in this case that's `axum-app`).
+
+Finally, deploy the app!
+
+```cmd
+fly launch
+```
+
+```output
+Scanning source code
+Detected a Dockerfile app
+Warning: This organization has no payment method, turning off high availability
+Creating app in [redacted]/fly-rust-axum-starter
+We're about to launch your app on Fly.io. Here's what you're getting:
```

Comment:
Do you think it would be a good idea to mention whether they should accept the defaults ('y' or 'n') or that they might want to change the name/region/other things?

Reply:
From a personal preference POV: I don't seen an immediate benefit to including it. My thoughts:

- I prefer short and to the point
- Once you get to that point, some y's or n's are not going to be impeding you

I'm however very open to arguments for why it would be good to include :)

Reply:
If you don't think they'll get hung up on anything there, then I trust you!